apache/airflow
# Add a migration script tooling support to migrate off mssql

Add a migration tool in order to deprecate MSSQL support in Airflow 2.8.0

I placed this in the `/scripts/tools` folder for now as it is not something targetted (yet) to be supported in main airflow core. Nevertheless because of the nature how it is it _could_ be made into one `airflow db ...` command.
I am open where we store it (for the moment).

How to test?
- Jump on any (dev or test) Airflow instance running 2.7.3, copy the python file `migrate_script.py` into a place where the Airflow python env is available
- Stop the runnign airflow processes and execute `python migrate_script.py --extract` and see that a `migration.db` SQLite file is built in your current folder.
- Copy the `migration.db` and the `migrate_script.py` to a test environment with Airflow 2.7.3, stop the airflow processes and and execute `python migrate_script.py --restore` - before hitting enter ensure no valuable data is contained, as it will wipe the DB content before copy!
- Start the Airflow processes and check that the content from the source system is visible in the UI. Check scheduler and logs of worker for errors. Start a DAG to see the system can run new stuff. Log-on with another user as well

I tested with:
- Source MSSQL, Target MySQL, Postgres and SQLite
- Example DAGs, execucted some runs, added a user, a variable and a pool

There might be other data "in the wild", probably there will be "glitches" if more situations are seen. Feedback welcome!
